projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-base.model.ts

```diff
@@ -101,6 +101,13 @@ export abstract class TableWidgetBaseModel {
   })
   public pageable: boolean = true;
 
+  @ModelProperty({
+    key: 'persistId',
+    displayName: 'ID used to enable and identify persisted table preferences',
+    type: STRING_PROPERTY.type
+  })
+  public persistId?: string; // Format: <page>.<table-name> (e.g. "explorer.spans-table")
```

Comment:
Hmmm don't love this, but I guess it's needed to identify the specific instance on the page. What are your thoughts to making this more generic and keeping this as a general model id (just in terms of naming)? We don't have to expose it on all models now, but would keep the concept consistent.

Reply:
So, we would want each model to have a separate `id` property, then we would change the `persistId` to just be a boolean for something like `enablePersist` or the like? We would then have a check to ensure there is a model ID set and use that as our key for the storage? 

Sounds fine.

Reply:
Are there specific tables we explicitly wouldn't want to persist settings for? If not, I'd leave out the boolean entirely and assume we can safely persist if there's an id present.

Reply:
I think just tree tables for something like services list, but that is more of a technical detail. Since column edit is already disabled for those types, and that is what triggers the persist we should be fine.

I'll assume all tables can persist then and we can back that assumption out and add the boolean if we encounter one that shouldn't be.

Reply:
Sure, I think we can build it out over time